### PR TITLE
[DEMO] pre-built charms as reusable artefacts in different GH workflow jobs

### PR DESCRIPTION
**Pull request by DnPlas**
Use data-platform-workloads' [build_with_cache](https://github.com/canonical/data-platform-workflows/blob/main/.github/workflows/build_charms_with_cache.yaml) action for speeding up charm builds and reuse charm files across different jobs.

### CODE REVIEW

File: tests/integration/helpers/localize_bundle.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def localize_bundle_application(
         if not resources:
             resources = get_resources_from_charm_dir(charm_dir)
 
-    bundle["applications"][application]["charm"] = str(charm_file)
+    bundle["applications"][application]["charm"] = f"./{charm_file}"
```
Comment (beliaev-maksim) [inline tests/integration/helpers/localize_bundle.py:76]: will it always be in the repo root ?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -48,22 +52,37 @@ jobs:
...
-    with:
-      charm-name: ${{ matrix.charm }}
+   steps:
+     - name: Download packed charm(s)
+       uses: actions/download-artifact@v3
+       with:
+         name: ${{ needs.build.outputs.artifact-name }}
+  
+      - name: Individual Integration Tests
+        needs: [quality-checks, unit-tests]
```
Comment (beliaev-maksim) [inline .github/workflows/ci.yaml:62]: should not we depend on build stage?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -13,6 +13,10 @@ on:
     - track/**
 
 jobs:
+  build:
```
Comment (beliaev-maksim) [inline .github/workflows/ci.yaml:16]: maybe more explicit ?
```suggestion
  build-charms:
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -48,22 +52,39 @@ jobs:
...
+       uses: actions/download-artifact@v3
+       with:
+         name: ${{ needs.build-charms.outputs.artifact-name }}
+  
+     - name: Individual Integration Tests
+       uses: canonical/charmed-kubeflow-workflows/.github/workflows/_integration-tests.yaml@dnplas-reusable-builds
+       secrets: inherit
+       with:
+         charm-name: ${{ matrix.charm }}
+         packed-charms: ${{ needs.build-charms.outputs.charms }}
```
Comment (beliaev-maksim) [inline .github/workflows/ci.yaml:72]:
```suggestion
     needs: [build-charms, quality-checks, unit-tests]
     matrix:
       charm:
         - kfp-api
         - kfp-profile-controller
     steps:
       - name: Download packed charm(s)
         uses: actions/download-artifact@v3
         with:
           name: ${{ needs.build-charms.outputs.artifact-name }}
    
       - name: Individual Integration Tests
         uses: canonical/charmed-kubeflow-workflows/.github/workflows/_integration-tests.yaml@dnplas-reusable-builds
         secrets: inherit
         with:
           charm-name: ${{ matrix.charm }}
           packed-charms: ${{ needs.build-charms.outputs.charms }}
```

File: charms/kfp-api/tests/integration/test_charm.py [hunk 5]
```diff
@@ -19,6 +20,24 @@
 MINIO_CONFIG = {"access-key": "minio", "secret-key": "minio-secret-key"}
 KFP_DB_CONFIG = {"database": "mlpipeline"}
 
+@pytest.fixture
+def ops_test(ops_test: OpsTest) -> OpsTest:
+    if os.environ.get("CI") == "true":
```
Comment (beliaev-maksim) [inline charms/kfp-api/tests/integration/test_charm.py:26]: this will fail if no ENV set

**Comment by DnPlas** [comment]
Closing as this PR has served its purpose (demo).

**Pull request closed without merging**